tsp/virtual_table_model.py

```diff
@@ -0,0 +1,124 @@
+# SOFTWARE.
+
+"""VirtualTableModel class file."""
+
+SIZE_KEY = "size"
+LOW_INDEX_KEY = "lowIndex"
+COLUMN_IDS_KEY = "columnIds"
+LINES_KEY = "lines"
+
+TABLE_LINE_INDEX_KEY = "index"
+TABLE_LINE_CELLS_KEY = "cells"
+TABLE_LINE_CELL_CONTENT_KEY = "content"
+
+
+# pylint: disable=too-few-public-methods
+class VirtualTableModel:
+    '''
+    Virtual table model that will be returned by the server
+    '''
+
+    def __init__(self, params):
+        # Size of the virtual table
+        self.size = 0
+        if SIZE_KEY in params:
+            if params.get(SIZE_KEY) is not None and type(params.get(SIZE_KEY)) is int:
+                self.size = int(params.get(SIZE_KEY))
+            del params[SIZE_KEY]
+
+        # Index of the first line in the virtual table
+        self.low_index = 0
+        if LOW_INDEX_KEY in params:
+            if params.get(LOW_INDEX_KEY) is not None and type(params.get(LOW_INDEX_KEY)) is int:
+                self.low_index = int(params.get(LOW_INDEX_KEY))
+            del params[LOW_INDEX_KEY]
+
+        # Array of column IDs in the virtual table
+        self.column_ids = []
+        if COLUMN_IDS_KEY in params:
+            if params.get(COLUMN_IDS_KEY) is not None:
+                for column_id in params.get(COLUMN_IDS_KEY):
+                    self.column_ids.append(column_id)
+            del params[COLUMN_IDS_KEY]
+
+        # Array of lines in the virtual table
+        self.lines = []
+        if LINES_KEY in params:
+            for line in params.get(LINES_KEY):
+                self.lines.append(VirtualTableLine(line))
+            del params[LINES_KEY]
+
+    def print(self):
+        print("VirtualTableModel:")
+        print(f"  size: {self.size}")
+        print(f"  low_index: {self.low_index}")
+        print(f"  column_ids: {self.column_ids}")
+
+        print("  lines:")
+        for i, line in enumerate(self.lines):
+            line.print()
+
+class VirtualTableLine:
+    '''
+    Virtual table line that will be returned by the server
+    '''
+
+    def __init__(self, params):
+        # Index of the line in the virtual table
+        self.index = -1
+        if TABLE_LINE_INDEX_KEY in params:
+            if params.get(TABLE_LINE_INDEX_KEY) is not None and type(params.get(TABLE_LINE_INDEX_KEY)) is int:
+                self.index = int(params.get(TABLE_LINE_INDEX_KEY))
+            del params[TABLE_LINE_INDEX_KEY]
+
+        # Array of cells in the line
+        self.cells = []
+        if TABLE_LINE_CELLS_KEY in params:
+            if params.get(TABLE_LINE_CELLS_KEY) is not None:
+                for cell in params.get(TABLE_LINE_CELLS_KEY):
+                    self.cells.append(VirtualTableLineCell(cell))
+            del params[TABLE_LINE_CELLS_KEY]
+
+    def print(self):
+
+        print(f"    index: {self.index}")
+        print("    cells:")
+        for i, cell in enumerate(self.cells):
+            cell.print()
+        print(f"    {'-' * 10}")
```

Comment:
There is also `tags` when the search_filter_expression is set to indicate that there is a match. `tags` is bitmask and right now we only support one non-null value if the cell matches. Maybe should print `matched` or `not matched` in case of searching. `tags` is an optional field and might not be present.

See here https://github.com/eclipse-tracecompass/org.eclipse.tracecompass/blob/d365dfc9a9d4305191387af7f8d078958e74d030/tmf/org.eclipse.tracecompass.tmf.core/src/org/eclipse/tracecompass/internal/provisional/tmf/core/model/events/TmfEventTableDataProvider.java#L569

Reply:
So based on my understanding, if tags==8 , then I should return matched? Or should I just return the tags itself?
Because based on the docs, it seems that 0 is none, 1/2 are reserved, 4 is border and 8 is highlight.